In [24]:
import pandas as pd

#1 load the foreign gifts data
df = pd.read_csv('ForeignGifts_edu.csv')
df




,ID,OPEID,Institution Name,City,State,Foreign Gift Received Date,Foreign Gift Amount,Gift Type,Country of Giftor,Giftor Name
0,1,102000,Jacksonville State University,Jacksonville,AL,43738,250000,Monetary Gift,CHINA,NaN
1,2,104700,Troy University,Troy,AL,43592,463657,Contract,CHINA,Confucius Institute Headquarters
2,3,105100,University of Alabama,Tuscaloosa,AL,43466,3649107,Contract,ENGLAND,Springer Nature Customer Service Ce
3,4,105100,University of Alabama,Tuscaloosa,AL,43472,1000,Contract,SAUDI ARABIA,Saudi Arabia Education Mission
4,5,105100,University of Alabama,Tuscaloosa,AL,43479,49476,Contract,SAUDI ARABIA,Saudi Arabia Education Mission
...,...,...,...,...,...,...,...,...,...,...
28216,28217,4279700,Albert Einstein College of Medicine,Bronx,NY,42704,381717,Contract,CHINA,Chia Tai TianQing Pharmaceutical Gr
28217,28218,4279700,Albert Einstein College of Medicine,Bronx,NY,42778,444938,Contract,ISRAEL,BL Oncology Ltd
28218,28219,4279700,Albert Einstein College of Medicine,Bronx,NY,42907,1064580,Contract,ENGLAND,Roche Products Limited
28219,28220,4279700,Albert Einstein College of Medicine,Bronx,NY,42948,737375,Contract,SWITZERLAND,F Hoffman-La Roche Ltd


In [ ]:
#2 describe the differences between different classes of gift types
df.groupby('Gift Type').describe()


ID                                                        \
                 count          mean          std     min       25%      50%   
Gift Type                                                                      
Contract       17274.0  14860.675408  8727.159706     2.0   7895.50  15892.5   
Monetary Gift  10936.0  12920.029901  6970.950105     1.0   6925.75  12179.5   
Real Estate       11.0  20888.818182  5664.792403  9246.0  23418.50  23421.0   

                                    OPEID                 ...  \
                    75%      max    count           mean  ...   
Gift Type                                                 ...   
Contract       22316.75  28220.0  17274.0  264930.739840  ...   
Monetary Gift  18119.25  28221.0  10936.0  223025.274323  ...   
Real Estate    23447.50  23459.0     11.0  300081.818182  ...   

              Foreign Gift Received Date          Foreign Gift Amount  \
                                     75%      max               count   
Gift Type                                                               
Contract                         43503.0  44012.0             17274.0   
Monetary Gift                    43403.0  43911.0             10936.0   
Real Estate                      43510.5  43646.0                11.0   

                                                                         \
                       mean           std       min       25%       50%   
Gift Type                                                                 
Contract       6.257945e+05  3.768217e+06 -537770.0   13757.5  110000.0   
Monetary Gift  5.281504e+05  2.086487e+06       1.0     250.0   50000.0   
Real Estate    1.335265e+06  1.412071e+06   39456.0  370574.5  667555.0   

                                      
                     75%         max  
Gift Type                             
Contract        366000.0  99999999.0  
Monetary Gift   400000.0  48842186.0  
Real Estate    2121736.0   4312000.0  

[3 rows x 32 columns]

There are three different Gift types, contract, monetary, and real estate. Contract has the highest count while 
real estate has the lowest count.Rreal estate has the highest min and monetary gift has the highest max.

In [11]:
df['Foreign Gift Amount'].groupby(df['Country of Giftor']).sum().sort_values(ascending=False)

Country of Giftor
QATAR           2706240869
ENGLAND         1464906771
CHINA           1237952112
SAUDI ARABIA    1065205930
BERMUDA          899593972
                   ...    
IRAN                    80
ST. LUCIA               50
CROATIA                 26
SLOVAKIA                20
ERITREA                  3
Name: Foreign Gift Amount, Length: 155, dtype: int64

Qatar gives the most money in total.

In [53]:
qatar_gifts = df[df['Country of Giftor'] == 'QATAR'].groupby('Institution Name')['Foreign Gift Amount'].sum().sort_values(ascending=False)
qatar_gifts  # Shows top 10 schools


Institution Name
Cornell University                            1018473315
Texas A&M University                           504099350
Carnegie Mellon University                     425410702
Georgetown University                          327128310
Northwestern University                        311148564
Virginia Commonwealth University                78867097
Houston Community College                        9438938
Harvard University                               5077754
University of Michigan - Ann Arbor               2878653
Arizona State University                         2515184
University of Southern California                2352800
Lamar University                                 2308232
Johns Hopkins University                         1917837
University of Colorado Denver                    1811405
Oregon State University                          1272923
Purdue University                                1245059
DePaul University                                1099213
University of 

In [52]:
df[df['Country of Giftor'] == 'QATAR'].shape[0]



693

In [10]:
df['Foreign Gift Amount'].groupby(df['Country of Giftor']).count().sort_values(ascending=False)

Country of Giftor
ENGLAND        3655
CHINA          2461
CANADA         2344
JAPAN          1896
SWITZERLAND    1676
               ... 
ERITREA           1
ARMENIA           1
ANTIGUA           1
PALAU             1
ROMANIA           1
Name: Foreign Gift Amount, Length: 155, dtype: int64

England initiates the most gifts by count.

In [12]:
df['Foreign Gift Amount'].groupby(df['Country of Giftor']).mean().sort_values(ascending=False)

Country of Giftor
BERMUDA      7.688837e+06
ECUADOR      3.985140e+06
QATAR        3.905109e+06
ANTIGUA      3.435545e+06
JERSEY       3.064589e+06
                 ...     
ST. LUCIA    5.000000e+01
IRAN         4.000000e+01
CROATIA      2.600000e+01
SLOVAKIA     2.000000e+01
ERITREA      3.000000e+00
Name: Foreign Gift Amount, Length: 155, dtype: float64

Bermuda gives the most gifts frequently on average

In [ ]:
df['Foreign Gift Amount'].groupby(df['Institution Name']).sum().sort_values(ascending=False)

Institution Name
Carnegie Mellon University               1477922504
Cornell University                       1289937761
Harvard University                        954803610
Massachusetts Institute of Technology     859071692
Yale University                           613441311
                                            ...    
Bates College                                250000
Metropolitan State University                235439
University of California, Merced             182514
University of Central Oklahoma               160288
University of Jamestown                         500
Name: Foreign Gift Amount, Length: 318, dtype: int64

Carnegie Mellon University has the most money in total

In [15]:
df['Foreign Gift Amount'].groupby(df['Institution Name']).count().sort_values(ascending=False)

Institution Name
University of California, Los Angeles          3916
California Institute of Technology             2502
Columbia University in the City of New York    1127
Ohio State University (The)                    1014
Johns Hopkins University                        899
                                               ... 
Post University                                   1
Curtis Institute of Music                         1
Washington Adventist University                   1
Corban University                                 1
Des Moines Area Community College                 1
Name: Foreign Gift Amount, Length: 318, dtype: int64

University of California, Los Angeles receives the most gifts by count

In [42]:
df.groupby('Institution Name')['Foreign Gift Amount'].mean().sort_values(ascending=False)


Institution Name
Brigham Young University - Idaho       2.270552e+07
Western International University       1.294216e+07
Whittier College                       1.207900e+07
University of Phoenix                  1.109435e+07
Curtis Institute of Music              1.000000e+07
                                           ...     
University of Alabama at Birmingham    2.983812e+04
University of California, Merced       1.520950e+04
Marquette University                   1.404990e+04
Westminster Theological Seminary       7.695829e+03
University of Jamestown                5.000000e+02
Name: Foreign Gift Amount, Length: 318, dtype: float64

In [20]:
round(df['Foreign Gift Amount'].median(), 2)

np.float64(94615.0)

median is 94,615

Why are they different?

Data would be right skewed if mean > median, meaning more high values

In [39]:
df['Foreign Gift Amount'].groupby([df['Country of Giftor'], df['Institution Name']]).sum().sort_values(ascending=False)


Country of Giftor  Institution Name                     
QATAR              Cornell University                       1018473315
BERMUDA            Carnegie Mellon University                750000000
QATAR              Texas A&M University                      504099350
                   Carnegie Mellon University                425410702
                   Georgetown University                     327128310
                                                               ...    
TURKEY             University of California, Merced                 10
UNITED NATIONS     Westminster Theological Seminary                 10
TRINIDAD           Westminster Theological Seminary                 10
ERITREA            University of California, Los Angeles             3
MEXICO             University of California, Merced                  1
Name: Foreign Gift Amount, Length: 2685, dtype: int64

Qatar to Cornell University was 1,018,473,315

In [32]:
df['Foreign Gift Amount'].groupby(df['Giftor Name']).count().sort_values(ascending=False)

Giftor Name
Saudi Arabian Cultural Mission    480
Human Frontier Science Program    287
Anonymous                         262
Qatar Foundation                  261
Novartis Pharmaceuticals Corp     251
                                 ... 
Illinois at Singapore Pte Ldt       1
Illinois at Singapore Pte LTD       1
Illinois at SIngapore Ptd Ltd       1
Illinios at Singapore Pte Ltd       1
LAU, C.K.                           1
Name: Foreign Gift Amount, Length: 8993, dtype: int64

The top three gifters were:
Saudi Arabian Cultural Mission
Human Frontier Science Program
Anonymous

Followed by..
Qatar Foundation
Novartis Pharmaceuticals Corp

In [ ]:
df.crosstab